api/tests/core/chronicles/test_api.py

```diff
@@ -45,3 +57,452 @@ def test_do_not_anonymize_chronicle_linked_to_user(self):
+                    text=ean,
+                ),
+                typeform.TypeformAnswer(
+                    field_id=constants.BookClub.CHRONICLE_ID.value,
+                    choice_id=None,
+                    text="some random chronicle description",
+                ),
+                typeform.TypeformAnswer(
+                    field_id=constants.BookClub.DIFFUSIBLE_PERSONAL_DATA_QUESTION_ID.value,
+                    choice_id=constants.BookClub.DIFFUSIBLE_PERSONAL_DATA_ANSWER_ID.value,
+                    text="oui",
+                ),
+                typeform.TypeformAnswer(
+                    field_id=constants.BookClub.SOCIAL_MEDIA_QUESTION_ID.value,
+                    choice_id=constants.BookClub.SOCIAL_MEDIA_ANSWER_ID.value,
+                    text="oui",
+                ),
+            ],
+        )
+
+        api.save_book_club_chronicle(form)
+
+        chronicle = models.Chronicle.query.first()
+        assert chronicle.age == 18
+        assert chronicle.city == "Paris"
+        assert chronicle.firstName == "Bob"
+        assert chronicle.ean == ean
+        assert chronicle.eanChoiceId == ean_choice_id
+        assert chronicle.content == "some random chronicle description"
+        assert chronicle.isIdentityDiffusible
+        assert chronicle.isSocialMediaDiffusible
+        assert not chronicle.isActive
+        assert chronicle.formId == form.response_id
+        assert chronicle.userId is None
+
+    def test_save_book_club_chronicle_empty(self):
+        ean = "1234567890123"
+        ean_choice_id = random_string()
+        form = typeform.TypeformResponse(
+            response_id=random_string(),
+            date_submitted=datetime.datetime(2024, 10, 24),
+            phone_number=None,
+            email="email@mail.test",
+            answers=[
+                typeform.TypeformAnswer(
+                    field_id=constants.BookClub.BOOK_EAN_ID.value,
+                    choice_id=ean_choice_id,
+                    text=ean,
+                ),
+                typeform.TypeformAnswer(
+                    field_id=constants.BookClub.CHRONICLE_ID.value,
+                    choice_id=None,
+                    text="some random chronicle description",
+                ),
+            ],
+        )
+
+        api.save_book_club_chronicle(form)
+
+        chronicle = models.Chronicle.query.first()
+        assert chronicle.age is None
+        assert chronicle.city is None
+        assert chronicle.firstName is None
+        assert chronicle.ean == ean
+        assert chronicle.eanChoiceId == ean_choice_id
+        assert chronicle.content == "some random chronicle description"
+        assert not chronicle.isIdentityDiffusible
+        assert not chronicle.isSocialMediaDiffusible
+        assert not chronicle.isActive
+        assert chronicle.formId == form.response_id
+        assert chronicle.userId is None
+
+    def test_save_book_club_chronicle_without_email(self):
```

Comment:
Détail : ce sera plus clair avec « do not save » (idem pour les suivants qui n'enregistrent pas)
```suggestion
    def test_do_not_save_book_club_chronicle_without_email(self):
```